primitives/consensus/babe/src/lib.rs

```diff
@@ -237,6 +244,87 @@ pub struct BabeEpochConfiguration {
 	pub allowed_slots: AllowedSlots,
 }
 
+/// Verifies the equivocation proof by making sure that: both headers have
+/// different hashes, are targetting the same slot, and have valid signatures by
+/// the same authority.
+pub fn check_equivocation_proof<H>(proof: EquivocationProof<H>) -> Option<()>
```

Comment:
API nit: a bit weird to use `Option` for this one. I'd expect either `Result<(), ()>` or `bool`